## <h1> PALMER PENGUIN DATA ANALYTICS </h1>
<hr>
<h2> Maranan, Reilh Rich P. </h2>
<h2> BSIT-3A </h2>

In [2]:
import numpy as np
import pandas as pd 
import warnings
warnings.filterwarnings('ignore')

<p>Importing dataset to Jupyter notebook </p>
<p>Converting dataset to dataframe </p>
<ul><strong>Reguirements</strong>
    <li>Name of the dataframe </li>
    <li>Source</li>
</ul>

In [3]:
penguins = pd.read_csv('C:\\Users\\User\\Desktop\\ELEC303\\Maranan, Reilh Rich P\\datasets\\penguins.csv')

In [4]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h1> CLEANING THE DATA FRAME <h1/>

In [5]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [6]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                333 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [7]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [8]:
penguins ['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [9]:
penguins ['species'] == 'Chinstrap'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: species, Length: 344, dtype: bool

In [10]:
penguins[penguins['species'] == 'Chinstrap']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [11]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [12]:
penguins['species'].unique()

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

In [13]:
Chinstrap_df = penguins[penguins['species'] == "Chinstrap"]

In [14]:
Adelie_df = penguins[penguins['species'] == "Adelie"]

In [15]:
Gentoo_df = penguins[penguins['species'] == "Gentoo"]

In [16]:
Gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [17]:
Adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [18]:
Chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [19]:
penguins['body_mass_g'] >= 4000

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Name: body_mass_g, Length: 344, dtype: bool

In [20]:
penguins[penguins['body_mass_g'] >= 4000]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
7       Adelie  Torgersen            39.2           19.6              195.0   
9       Adelie  Torgersen            42.0           20.2              190.0   
14      Adelie  Torgersen            34.6           21.1              198.0   
17      Adelie  Torgersen            42.5           20.7              197.0   
19      Adelie  Torgersen            46.0           21.5              194.0   
..         ...        ...             ...            ...                ...   
323  Chinstrap      Dream            49.0           19.6              212.0   
329  Chinstrap      Dream            50.7           19.7              203.0   
333  Chinstrap      Dream            49.3           19.9              203.0   
339  Chinstrap      Dream            55.8           19.8              207.0   
342  Chinstrap      Dream            50.8           19.0              210.0   

     body_mass_g   sex  year  
7         4675.0  male  2007  
9         4250.0   NaN  2007  
14        4400.0  male  2007  
17        4500.0  male  2007  
19        4200.0  male  2007  
..           ...   ...   ...  
323       4300.0  male  2009  
329       4050.0  male  2009  
333       4050.0  male  2009  
339       4000.0  male  2009  
342       4100.0  male  2009  

[177 rows x 8 columns]

In [21]:
(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000)

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Length: 344, dtype: bool

In [22]:
Chinstrap_Heavy = [(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000)]

In [23]:
Chinstrap_Heavy

[0      False
 1      False
 2      False
 3      False
 4      False
        ...  
 339     True
 340    False
 341    False
 342     True
 343    False
 Length: 344, dtype: bool]

In [24]:
penguins ['island']

0      Torgersen
1      Torgersen
2      Torgersen
3      Torgersen
4      Torgersen
         ...    
339        Dream
340        Dream
341        Dream
342        Dream
343        Dream
Name: island, Length: 344, dtype: object

In [25]:
penguins ['island'] == 'Dream'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: island, Length: 344, dtype: bool

In [26]:
penguins[penguins ['island'] == 'Dream']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30      Adelie  Dream            39.5           16.7              178.0   
31      Adelie  Dream            37.2           18.1              178.0   
32      Adelie  Dream            39.5           17.8              188.0   
33      Adelie  Dream            40.9           18.9              184.0   
34      Adelie  Dream            36.4           17.0              195.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
30        3250.0  female  2007  
31        3900.0    male  2007  
32        3300.0  female  2007  
33        3900.0    male  2007  
34        3325.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[124 rows x 8 columns]

In [27]:
penguins ['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

In [28]:
penguins['island'].unique()

array(['Torgersen', 'Biscoe', 'Dream'], dtype=object)

In [29]:
Dream_df = penguins[penguins ['island'] == "Dream"]

In [30]:
Dream_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30      Adelie  Dream            39.5           16.7              178.0   
31      Adelie  Dream            37.2           18.1              178.0   
32      Adelie  Dream            39.5           17.8              188.0   
33      Adelie  Dream            40.9           18.9              184.0   
34      Adelie  Dream            36.4           17.0              195.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
30        3250.0  female  2007  
31        3900.0    male  2007  
32        3300.0  female  2007  
33        3900.0    male  2007  
34        3325.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[124 rows x 8 columns]

In [31]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [32]:
penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

0      39.10000
1      39.50000
2      40.30000
3      43.92193
4      36.70000
         ...   
339    55.80000
340    43.50000
341    49.60000
342    50.80000
343    50.20000
Name: bill_length_mm, Length: 344, dtype: float64

In [33]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [34]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                333 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [35]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [36]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [37]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        2
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [38]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

<h1> HOW TO CHANGE THE DATA TYPE OF A COLUMN </h1>

In [39]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [40]:
penguins['species'] = penguins['species'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')
penguins['island'] = penguins['island'].astype('category')

In [41]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      342 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    int64   
dtypes: category(3), float64(4), int64(1)
memory usage: 14.9 KB


<h1> HOW TO CREATE A COLUMN </h1>

In [42]:
penguins['mass_kg'] = np.round (penguins['body_mass_g'] / 1000,1)

In [43]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000           18.7         181.000000   
1       Adelie  Torgersen        39.50000           17.4         186.000000   
2       Adelie  Torgersen        40.30000           18.0         195.000000   
3       Adelie  Torgersen        43.92193            NaN         200.915205   
4       Adelie  Torgersen        36.70000           19.3         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000           19.8         207.000000   
340  Chinstrap      Dream        43.50000           18.1         202.000000   
341  Chinstrap      Dream        49.60000           18.2         193.000000   
342  Chinstrap      Dream        50.80000           19.0         210.000000   
343  Chinstrap      Dream        50.20000           18.7         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

<h1> EXPLORATORY DATA ANALYTICS (EDA) </h1>

<h1> 1. WHICH SPECIES HAS THE LONGEST IN FLIPPER LENGTH MM ? </h1>

In [44]:
Chinstrap_df['flipper_length_mm'].mean()

np.float64(195.8235294117647)

In [45]:
Adelie_df['flipper_length_mm'].mean()

np.float64(189.95364238410596)

In [46]:
Gentoo_df['flipper_length_mm'].mean()

np.float64(217.1869918699187)

<h1> INSIGHT #1 <h/1>

<h3> THE LONGEST SPECIES FROM DATASET IS THE GENTOO SPECIES</h3>

<hr>

<h1>2. WHICH SPECIES HAS THE HIGHEST AVERAGE BODY MASS G? </h1>

In [47]:
Chinstrap_df['body_mass_g'].mean()

np.float64(3733.0882352941176)

In [48]:
Adelie_df['body_mass_g'].mean()

np.float64(3700.662251655629)

In [49]:
Gentoo_df['body_mass_g'].mean()

np.float64(5076.016260162602)

<h1> INSIGHT #2 </h1>

<h2> THE HIGHEST SPECIES FROM DATASETS IS THE GENTOO SPECIES </h2>

<hr>

<h1> 3. WHICH ISLAND HAS THE MOST PENGUINS RECORDED? </h1>

In [50]:
penguins['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h1> INSIGHT #3 </h1>

<h2> THE ISLAND WITH THE MOST PENGUIN RECORD IS BISCOE </h2>

<hr>

<h1> 4. WHICH SPECIES HAS THE SMALLEST AVERAGE BILL DEPTH (MM)?  </h1>

In [51]:
Chinstrap_df['bill_depth_mm'].mean()

np.float64(18.42058823529412)

In [52]:
Adelie_df['bill_depth_mm'].mean()

np.float64(18.346357615894043)

In [53]:
Gentoo_df['bill_depth_mm'].mean()

np.float64(14.982113821138212)

<h1> INSIGHT #4 </h1>

<h2> THE SPECIES WITH THE SMALLEST AVERAGE BILL DEPTH IS THE GENTOO SPECIES </h2>

<hr>

<h1> 5. HOW MANY MALE AND FEMALE PENGUINS ARE THERE IN THE DATASET? </h1>

In [54]:
penguins['sex'].value_counts()

sex
male      179
female    165
Name: count, dtype: int64

<h1> INSIGHT #5 </h1>

<h2> THERE ARE 168 MALE AND 165 FEMALE PENGUINS IN THE DATASET </h2>

<hr>

<h1> 6. WHICH YEAR RECORDED THE MOST PENGUIN ENTRIES </h1>

In [55]:
penguins['year'].value_counts()

year
2009    120
2008    114
2007    110
Name: count, dtype: int64

<h1> INSIGHT #6 </h1>

<h2> THE YEAR 2009 HAD THE MOST PENGUIN ENTRIES </h2>

<hr>

<h1> 7. WHAT IS THE AVERAGE FLIPPER LENGTH FOR MALE VS FEMALE PENGUINS? </h1>

In [56]:
penguins.groupby('sex')['flipper_length_mm'].mean()

sex
female    197.363636
male      204.188997
Name: flipper_length_mm, dtype: float64

<h1> INSIGHT #7 </h1>

<h2> ON AVERAGE, MALE PENGUINS HAVE LONGER FLIPPERS </h2> 

<hr>

<h1> DATA AGGREGATION </h1>

<h1> 8. WHAT IS THE BILL LENGTH OF EACH SPECIES? </h1>

In [58]:
avg_bill_length = penguins.groupby('species') ['bill_length_mm'].mean()

In [59]:
avg_bill_length

species
Adelie       38.825144
Chinstrap    48.833824
Gentoo       47.475983
Name: bill_length_mm, dtype: float64

<h1> INSIGHT #8 </h1>

<h3> THE HIGHEST BILL LENGTH IS THE CHINSTRAP SPECIES </h3>

<hr>

<h1> 9. WHAT IS THE LONGEST FLIPPER LENGTH PER ISLAND? </h1>

In [60]:
max_flipper_length = penguins.groupby('island') ['flipper_length_mm'].max()

In [61]:
max_flipper_length

island
Biscoe       231.0
Dream        212.0
Torgersen    210.0
Name: flipper_length_mm, dtype: float64

<h1> INSIGHT #9 </h1>

<h3> THE LONGEST FLIPPER LENGTH IS THE BISCOE ISLAND </h3>

<hr>

<h1> 10. WHAT IS THE BIGGEST SPECIES IN ALL ISLAND? </h1>

In [64]:
max_body_mass = penguins.groupby('species') ['body_mass_g'].max()

In [65]:
max_body_mass

species
Adelie       4775.0
Chinstrap    4800.0
Gentoo       6300.0
Name: body_mass_g, dtype: float64

<h1> INSIGHT #10 </h1>

<h3> GENTOO IS THE BIGGEST IN ALL ISLAND </h3> 

<hr>

<h1> TESTING CORRELATION </h1>

<h3> PEARSON R CORRELATION </h3>
<hr>
<p> USED FOR TESTING CORRELATION BETWEEN TWO <strong> NUMERICAL VALUES </strong> </p>

<h1> 11. IS THERE A RELATIONSHIP BETWEEN A PENGUINS'S BILL LENGTH AND BILL DEPTH? </h1>

In [66]:
bill_length_depth_cor = penguins['bill_length_mm'].corr(penguins['bill_depth_mm'])

In [67]:
bill_length_depth_cor

np.float64(-0.23505287035553257)

<h1> INSIGHT #11 </h1>

<h3> IT'S NEGATIVE, THEREFORE IT MEANS THAT PENGUINS WITH LONGER BILLS USUALLY HAVE SHALLOWER BILLS </h3>

<hr>

<h1> 12. IS THERE A RELATIONSHIP BETWEEN PENGUINS'S BODY MASS AND BILL LENGTH? </h1>

In [68]:
body_bill_cor = penguins['body_mass_g'].corr(penguins['bill_length_mm'])

In [69]:
body_bill_cor

np.float64(0.5951098244376305)

<h1> INSIGHT #12 </h1>

<h3> IT'S POSITIVE, THEREFORE IT MEANS THAT PENGUINS WITH LONGER FLIPPERS USUALLY HAVE LONGER BILLS </h3>

<hr>